Simplify code for readability and debugging

lucc opened the pull request:
When I tried to look into #1 I found these improvements (but not yet an actual fix).

lucc committed:
Remove unneeded `:exe` wrappings

lucc committed:
Add gitignore file

lucc committed:
Simplify :substitute commands

If the whole line was selected a restriction of the substitution to the
selected are is not necessary.

lucc committed:
Improve global replacements of underlines

Originally the position of the cursor was kept and only the remainder of the
underline from the cursor position to the end of line replaced.

lucc committed:
Remove unnecessary cursor movement

lucc committed:
Simplify some substitutions

lucc committed:
Use :substitute to put | around links

lucc committed:
Remove many :silent modifiers to ease debugging

The messages should still be hidden in normal usage because the relevant
functions are called with :silent.  But now the function can manually be
called without :silent to see the messages.

lucc committed:
Split search and selection to remove :exe

lucc committed:
Abstract common code into a function

lucc committed:
Use printf() instead of manuall formatting

lucc committed:
Avoid :split without args

It is not necessary to `:split` in order to `:edit file`, one can just
`:split file`.  And with certain setups the former opens the last edited file
in the original half of the split if no file was yet opened (for example with
`vim -c 'Vinfo something'`).

alx741 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+doc/tags
+plugin/vinfo_doc_repo/
```
Can't quite remember how this was supposed to work, but i think the plugin will fail if it can't find the `plugin/vinfo_doc_repo` directory, and it can't create it. Have you test that?

alx741 commented:
This is great! thanks, i left an annotation there, please check it out.

lucc committed:
Move generated files to xdg compilant location

The generated help files will be in ${XDG_DATA_HOME:-~/.local/share}/vinfo.

lucc commented:
I did not see that the directory was tracked with git and did not test further. I just assumed that it would be created via `mkdir -p` on the fly. Sorry.

I added 3ec9c711c28d09037724ba31900ae1e7cf66c30c to move the location of the generated files. This also solves the problem of the previous comment. I also changed the gitignore commit.

alx741 commented:
Awesome! thanks
